[RFC] apps:glusterfs: make it possible to get 1xN volumes (N=#replica)

obnoxxx opened the pull request:
Start the BrickSizeGenerator with the full volume size.

Signed-off-by: Michael Adam <obnox@redhat.com>

obnoxxx commented:
Note: [RFC] tag in the commit msg. Should we decide to push this I would remove that tag from the commit msg... ;-)

lpabon commented on apps/glusterfs/volume_durability_test.go:
```diff
@@ -89,32 +89,39 @@ func TestNoneDurability(t *testing.T) {
 	sets, brick_size, err := gen()
 	tests.Assert(t, err == nil)
+	tests.Assert(t, sets == 1)
+	tests.Assert(t, brick_size == 100*GB)
```
awesome

lpabon commented:
ok to test @centos-ci

obnoxxx commented:
I have no idea what's wrong with the centos-ci.

The error message is "Build finished." ... !?!

And the output seems to indicate that all went well, except it reports FAIL at the end:

```
[negroni] Completed 200 OK in 2.123016ms
PASS
ok  	github.com/heketi/heketi/tests/functional/TestVolumeSnapshotBehavior/tests	116.920s
Shutting down...
[heketi] INFO 2016/09/08 19:54:43 Closed
==> storage3: Removing domain...
==> storage2: Removing domain...
==> storage1: Removing domain...
==> storage0: Removing domain...
==> PASSED TestVolumeSnapshotBehavior
==> Started Thu Sep  8 18:38:31 BST 2016
==> Ended Thu Sep  8 19:54:51 BST 2016
==> FAILED

This test is part of the Gluster Jenkins jobs that can be found on GitHub:
 - https://github.com/gluster/glusterfs-patch-acceptance-tests/tree/master/centos-ci

Build step 'Execute shell' marked build as failure
Setting status of bd73dbcee65df5c690671a3c451e41c5ec1886be to FAILURE with url https://ci.centos.org/job/gluster_heketi-functional/65/ and message: 'Build finished. '
Using context: centos-ci
Finished: FAILURE
```

lpabon commented:
@obnoxxx i'll take a look at the ci

lpabon commented:
The issue is this:

```
--- FAIL: TestHeketiVolumes (455.50s)
	assert.go:44: 
		
ASSERT:	func (github.com/heketi/heketi/tests/functional/TestSmokeTest/tests.TestHeketiVolumes) 0x48629b
		
	File /root/go/src/github.com/heketi/heketi/tests/functional/TestSmokeTest/tests/heketi_test.go:228
FAIL
exit status 1
FAIL	github.com/heketi/heketi/tests/functional/TestSmokeTest/tests	455.515s
godep: go exit status 1
```

obnoxxx commented:
oh, thanks -- where did you see that error message? I did not find that in jenkins..

obnoxxx commented:
Adapted the test. Re-running functional test locally before resubmitting...

lpabon commented:
@obnoxxx If you click on `Details`, then click on `Console Output`, then click on `Full Log`, you can search for the word `fail`.  Some failures are expected, so do not pay attention to those :smile:

obnoxxx commented:
strange, i've been running it here on my system, and it keeps succeeding witht Bricks == 6 and failing with Bricks == 3 ... :-o

obnoxxx commented:
Ok. Now I understood why it always reported 6 bricks:
it was testing the old code.

I created issue #496 and provided a fix for that in pr #497....
Updated patche following

obnoxxx commented:
I updated the patchset.

@lpabon warning: I again updated the original patch, i.e. squashed the change to the functional test into the original commit. And I added a new small patch that adds the logging of the brick number to the assert statement.

obnoxxx commented:
retest this please @centos-ci
